src/main/java/org/broadinstitute/dsm/route/BSPKitQueryRoute.java

```diff
@@ -2,157 +2,39 @@
-            return new BSPKitStatus(BSPKitStatus.EXITED);
-        }
-        if (StringUtils.isNotBlank(bspKitInfo.getDeactivationDate())) {
-            return new BSPKitStatus(BSPKitStatus.DEACTIVATED);
-        }
-        if (StringUtils.isNotBlank(bspKitInfo.getDdpParticipantId())) {
-            DDPInstance ddpInstance = DDPInstance.getDDPInstance(bspKitInfo.getInstanceName());
-            InstanceSettings instanceSettings = new InstanceSettings();
-            InstanceSettingsDto instanceSettingsDto = instanceSettings.getInstanceSettings(bspKitInfo.getInstanceName());
-            Value received = instanceSettingsDto
-                    .getKitBehaviorChange()
-                    .map(kitBehavior -> kitBehavior.stream().filter(o -> o.getName().equals(InstanceSettings.INSTANCE_SETTING_RECEIVED)).findFirst().orElse(null))
-                    .orElse(null);
-
-            if (received != null && StringUtils.isNotBlank(ddpInstance.getParticipantIndexES())) {
-                Map<String, Map<String, Object>> participants = ElasticSearchUtil.getFilteredDDPParticipantsFromES(ddpInstance,
-                    ElasticSearchUtil.BY_GUID + bspKitInfo.getDdpParticipantId());
-                writeSampleReceivedToES(ddpInstance, bspKitInfo);
-                Map<String, Object> participant = participants.get(bspKitInfo.getDdpParticipantId());
-                if (participant != null) {
-                    boolean specialBehavior = InstanceSettings.shouldKitBehaveDifferently(participant, received);
-                    if (specialBehavior) {
-                        //don't trigger ddp to sent out email, only email to study staff
-                        if (InstanceSettings.TYPE_NOTIFICATION.equals(received.getType())) {
-                            String message = "Kit of participant " + bspKitInfo.getBspParticipantId() + " was received by GP. <br> " +
-                                    "CollaboratorSampleId:  " + bspKitInfo.getBspSampleId() + " <br> " +
-                                    received.getValue();
-                            notificationUtil.sentNotification(bspKitInfo.getNotificationRecipient(), message, NotificationUtil.UNIVERSAL_NOTIFICATION_TEMPLATE, NotificationUtil.DSM_SUBJECT);
-                        }
-                        else {
-                            logger.error("Instance settings behavior for kit was not known " + received.getType());
-                        }
-                    }
-                    else {
-                        triggerDDP(conn, bspKitInfo, firstTimeReceived, kitLabel);
-                    }
-                }
-            }
-            else {
-                triggerDDP(conn, bspKitInfo, firstTimeReceived, kitLabel);
+        BSPKit bspKit = new BSPKit();
+        if (!bspKit.canReceiveKit(kitLabel)) {
+            Optional<Object> result = Optional.ofNullable(bspKit.getKitStatus(kitLabel, notificationUtil));
```

Comment:
Instead of wrapping result of `getKitStatus` here, as mentioned above, make method `getKitStatus` to return Optional is better.